src/main/java/homework_6/map_problems_generator/Main.java

```diff
@@ -0,0 +1,22 @@
+package homework_6.map_problems_generator;
+
+import java.util.HashMap;
+
+public class Main {
+    public static void main(String[] args) {
+        HashMap<MapProblemsCollisionGenerator<String, String>, String> container_problems_coll_gen = new HashMap<>();
+        container_problems_coll_gen.put(new MapProblemsCollisionGenerator<>("one", "one"), "one");
+        container_problems_coll_gen.put(new MapProblemsCollisionGenerator<>("two", "two"), "two");
+
+        String resultTwo_1 = container_problems_coll_gen.get(new MapProblemsCollisionGenerator<>("two", "two"));
+
+
+
+        HashMap<MapProblemsMutableGenerator<String>, String> container_mutable_gen = new HashMap<>();
```

Comment:
There's no mutable key problem in your example
![image](https://user-images.githubusercontent.com/58872509/134233036-96a37b21-0855-4133-8c50-2e515ed13136.png)

Reply:
Fixed!